Extension/src/common.ts

```diff
@@ -1003,6 +1018,9 @@ export function extractCompilerPathAndArgs(inputCompilerPath?: string, inputComp
                 return inputCompilerArgs.indexOf(item) < 0;
             }));
     }
+    if (compilerName) {
+        compilerName = compilerName.toLowerCase();
```

Comment:
This makes compilerName lower case on all platforms.  But, it looks like it's being compared against a path in getTasks() in cppBuildTaskProvider.ts, which would seem to cause a bug.  It may be safer to keep as the original case, and convert to lower case when comparing against "cl.exe".